libcontainer/intelrdt/intelrdt.go

```diff
@@ -159,10 +159,23 @@ func NewManager(config *configs.Config, id string, path string) *Manager {
 	if config.IntelRdt == nil {
 		return nil
 	}
-	if _, err := Root(); err != nil {
-		// Intel RDT is not available.
+
+	rootPath, err := Root()
+	if err != nil {
 		return nil
 	}
+	// NOTE: Should we check if the path provided as arg matches the path
+	// constructed below? If not, we're screwed as we've effectively lost resctrl
+	// control of the container (e.g. because the resctrl fs was unmounted or
+	// remounted elsewhere). All operations are deemed to fail.
+	if path == "" {
+		clos := id
+		if config.IntelRdt.ClosID != "" {
+			clos = config.IntelRdt.ClosID
+		}
+		path = filepath.Join(rootPath, clos)
```

Comment:
I know this is just moved from the previous code, but we need to avoid `filepath.Join` in runc as it is vulnerable to all sorts of attacks.

I think we should at least be using `securejoin.SecureJoin` here...

Reply:
Thx for the review @cyphar. Changed as suggested.